llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -1062,13 +1062,16 @@ class LowerMatrixIntrinsics {
       Value *Op1;
       Value *Op2;
       if (auto *BinOp = dyn_cast<BinaryOperator>(Inst))
-        Changed |= VisitBinaryOperator(BinOp);
+        VisitBinaryOperator(BinOp);
       if (auto *UnOp = dyn_cast<UnaryOperator>(Inst))
-        Changed |= VisitUnaryOperator(UnOp);
+        VisitUnaryOperator(UnOp);
       if (match(Inst, m_Load(m_Value(Op1))))
-        Changed |= VisitLoad(cast<LoadInst>(Inst), Op1, Builder);
+        VisitLoad(cast<LoadInst>(Inst), Op1, Builder);
       else if (match(Inst, m_Store(m_Value(Op1), m_Value(Op2))))
-        Changed |= VisitStore(cast<StoreInst>(Inst), Op1, Op2, Builder);
+        VisitStore(cast<StoreInst>(Inst), Op1, Op2, Builder);
+      else
+        continue;
+      Changed = true;
```

Comment:
I don't think the `Changed` logic is still correct, unless you make it a single if/else chain. With the current setup, binops will hit the `continue` path.